### DISCUSSION TITLE
Features corresponding to new features in Midjourney v6

### DISCUSSION THREAD

**Discussion by IamTirion** [lllyasviel/Fooocus#2250] [Ideas]
If I'm not mistaken, Fooocus is an attempt to replicate Midjourney. There are quite a few new features in Midjourney v6, such as Style Reference. Will those be added to Fooocus?

Edit: Let me be a bit clearer. I think the new features are namely Style Reference, and Zoom out. I have explained Style Reference below. Zoom out is also unique. It is not outpainting. It is like zooming out in SVD. If you have an image on the close up of a face, you can generate another image with the whole body. Basically Midjourney v6 is all about attempting to achieve consistency with image generation.

**Comment by mashb1t** [comment]
You can use Image Prompt to reference styles if you refer to this, but currently there is no matrix decision feature to build your own Mini-LoRA / style reference.

**Comment by IamTirion** [comment]
> You can use Image Prompt to reference styles if you refer to this, but currently there is no matrix decision feature to build your own Mini-LoRA / style reference.

It's quite different. The Fooocus Image Prompt is based on the Midjourney Image Prompt, which itself is based on Img2Img. It copies more than just the style of the input image. The new Style Reference feature in Midjourney is more like making images in the same setting/environment. It is an attempt towards consistent image generation.

**Reply by mashb1t** [reply]
i got what it is, so you basically are referring to https://github.com/lllyasviel/Fooocus/issues/1199 or similar, correct?

**Reply by IamTirion** [reply]
> i got what it is, so you basically are referring to #1199 or similar, correct?

No. That one is still "too similar" to the input image. Take a look at this at the 2:53 timestamp:
https://youtu.be/ojIZJtP3cnU?t=173]

You can see that it managed to produce images that feel like they are from the same forest or crash site with different camera angles. StyleAlign, if I have to describe it, feels like it produces images with the "same" background if you understand what I mean.
